packages/orchestration/src/orchestration-embedding-client.test.ts

```diff
@@ -0,0 +1,304 @@
+        masking_providers: [
+          buildDpiMaskingProvider({
+            method: 'pseudonymization',
+            entities: ['profile-person', 'profile-location']
+          })
+        ]
+      }
+    };
+
+    const request: EmbeddingRequest = {
+      input:
+        'My name is Tom. I am applying as a Senior Software Dev. I work closely with Jerry.'
+    };
+
+    const mockResponse = await parseMockResponse<EmbeddingsPostResponse>(
+      'orchestration',
+      'orchestration-embedding-with-masking-response.json'
+    );
+
+    mockInference(
+      {
+        data: constructEmbeddingPostRequest(config, request)
+      },
+      {
+        data: mockResponse,
+        status: 200
+      },
+      {
+        url: 'inference/deployments/1234/v2/embeddings'
+      }
+    );
+
+    const response = await new OrchestrationEmbeddingClient(config).embed(
+      request
+    );
+
+    expect(response).toBeInstanceOf(OrchestrationEmbeddingResponse);
+    expect(response._data.request_id).toBe('random-request-id');
+    expect(response.getIntermediateResults()).toBeDefined();
+    expect(response.getIntermediateResults()?.input_masking).toBeDefined();
+    expect(
+      response.getIntermediateResults()?.input_masking?.data?.masked_input
+    ).toContain(
+      'My name is MASKED_PERSON. I am applying as a Senior Software Dev. I work closely with MASKED_PERSON.'
+    );
+    expect(response.getEmbeddingVectors()).toEqual(expect.any(Array));
+  });
+
+  it('executes a request with the custom resource group', async () => {
+    const config: EmbeddingModuleConfig = {
+      embeddings: {
+        model: {
+          name: 'text-embedding-3-small'
+        }
+      }
+    };
+
+    const request: EmbeddingRequest = {
+      input: 'Test text for embedding with custom resource group'
+    };
+
+    const customEmbeddingEndpoint = {
+      url: 'inference/deployments/1234/v2/embeddings',
+      resourceGroup: 'custom-resource-group'
+    };
+
+    const mockResponse = await parseMockResponse<EmbeddingsPostResponse>(
+      'orchestration',
+      'orchestration-embedding-with-masking-response.json'
+    );
+
+    mockDeploymentsList(
+      { scenarioId: 'orchestration', resourceGroup: 'custom-resource-group' },
+      {
+        id: '1234',
+        model: { name: 'text-embedding-3-small', version: 'latest' }
+      }
+    );
+
+    mockInference(
+      {
+        data: constructEmbeddingPostRequest(config, request)
+      },
+      {
+        data: mockResponse,
+        status: 200
+      },
+      customEmbeddingEndpoint
+    );
+
+    const clientWithResourceGroup = new OrchestrationEmbeddingClient(config, {
+      resourceGroup: 'custom-resource-group'
+    });
+
+    const response = await clientWithResourceGroup.embed(request);
+    expect(response).toBeInstanceOf(OrchestrationEmbeddingResponse);
+    expect(response._data.request_id).toBe('random-request-id');
+  });
+
+  it('handles HTTP error responses', async () => {
```

Comment:
[q] What is the point of this test? It is basically testing if Axios throws when response is 400. And since 400 is mocked, it is not relevant to the given `config`, `request` or `constructEmbeddingPostRequest()`. Can you clarify what is under test or we just remove this?